File: src_backup_phpstan/Entity/Offres/Offre.php
<?php

namespace App\Entity\Offres;

use App\Repository\OffreRepository;
use Doctrine\ORM\Mapping as ORM;

#[ORM\Entity(repositoryClass: OffreRepository::class)]
#[ORM\Table(name: 'offre')]
class Offre
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column(name: 'id_offre', type: 'integer')]
    private ?int $idOffre = null;

    #[ORM\Column(name: 'titre', type: 'string', length: 200)]
    private ?string $titre = null;

    #[ORM\Column(name: 'description', type: 'text', nullable: true)]
    private ?string $description = null;

    #[ORM\Column(name: 'type_offre', type: 'string', length: 50, nullable: true)]
    private ?string $typeOffre = null;

    #[ORM\Column(name: 'budget', type: 'decimal', precision: 12, scale: 2, nullable: true)]
    private ?string $budget = null;

    #[ORM\Column(name: 'prix', type: 'decimal', precision: 12, scale: 2, nullable: true)]
    private ?string $prix = null;

    #[ORM\Column(name: 'duree', type: 'integer', nullable: true)]
    private ?int $duree = null;

    #[ORM\Column(name: 'matricule', type: 'string', length: 50, nullable: true)]
    private ?string $matricule = null;

    #[ORM\Column(name: 'localisation', type: 'string', length: 150, nullable: true)]
    private ?string $localisation = null;

    #[ORM\Column(name: 'date_publication', type: 'date', nullable: true)]
    private ?\DateTimeInterface $datePublication = null;

    #[ORM\Column(name: 'date_limite', type: 'date', nullable: true)]
    private ?\DateTimeInterface $dateLimite = null;

    #[ORM\Column(name: 'statut', type: 'string', length: 50, nullable: true)]
    private ?string $statut = 'ouverte';

    #[ORM\Column(name: 'id_utilisateur', type: 'integer', nullable: true)]
    private ?int $idUtilisateur = null;

    #[ORM\Column(name: 'mail_user', type: 'string', length: 150, nullable: true)]
    private ?string $mailUser = null;

    #[ORM\Column(name: 'vues', type: 'integer', nullable: true)]
    private ?int $vues = 0;

    #[ORM\Column(name: 'tags', type: 'string', length: 500, nullable: true)]
    private ?string $tags = null;

    public function getIdOffre(): ?int
    {
        return $this->idOffre;
    }

    public function getId(): ?int
    {
        return $this->idOffre;
    }

    public function getTitre(): ?string
    {
        return $this->titre;
    }

    public function setTitre(?string $titre): self
    {
        $this->titre = $titre;
        return $this;
    }

    public function getDescription(): ?string
    {
        return $this->description;
    }

    public function setDescription(?string $description): self
    {
        $this->description = $description;
        return $this;
    }

    public function getTypeOffre(): ?string
    {
        return $this->typeOffre;
    }

    public function setTypeOffre(?string $typeOffre): self
    {
        $this->typeOffre = $typeOffre;
        return $this;
    }

    public function getBudget(): ?string
    {
        return $this->budget;
    }

    public function setBudget($budget): self
    {
        $this->budget = $budget !== null ? (string) $budget : null;
        return $this;
    }

    public function getPrix(): ?string
    {
        return $this->prix;
    }

    public function setPrix($prix): self
    {
        $this->prix = $prix !== null ? (string) $prix : null;
        return $this;
    }

    public function getDuree(): ?int
    {
        return $this->duree;
    }

    public function setDuree(?int $duree): self
    {
        $this->duree = $duree;
        return $this;
    }

    public function getMatricule(): ?string
    {
        return $this->matricule;
    }

    public function setMatricule(?string $matricule): self
    {
        $this->matricule = $matricule;
        return $this;
    }

    public function getLocalisation(): ?string
    {
        return $this->localisation;
    }

    public function setLocalisation(?string $localisation): self
    {
        $this->localisation = $localisation;
        return $this;
    }

    public function getDatePublication(): ?\DateTimeInterface
    {
        return $this->datePublication;
    }

    public function setDatePublication(?\DateTimeInterface $datePublication): self
    {
        $this->datePublication = $datePublication;
        return $this;
    }

    public function getDateLimite(): ?\DateTimeInterface
    {
        return $this->dateLimite;
    }

    public function setDateLimite(?\DateTimeInterface $dateLimite): self
    {
        $this->dateLimite = $dateLimite;
        return $this;
    }

    public function getStatut(): ?string
    {
        return $this->statut;
    }

    public function setStatut(?string $statut): self
    {
        $this->statut = $statut;
        return $this;
    }

    public function getIdUtilisateur(): ?int
    {
        return $this->idUtilisateur;
    }

    public function setIdUtilisateur(?int $idUtilisateur): self
    {
        $this->idUtilisateur = $idUtilisateur;
        return $this;
    }

    public function getMailUser(): ?string
    {
        return $this->mailUser;
    }

    public function setMailUser(?string $mailUser): self
    {
        $this->mailUser = $mailUser;
        return $this;
    }

    public function getVues(): ?int
    {
        return $this->vues;
    }

    public function setVues(?int $vues): self
    {
        $this->vues = $vues;
        return $this;
    }

    public function getTags(): ?string
    {
        return $this->tags;
    }

    public function setTags(?string $tags): self
    {
        $this->tags = $tags;
        return $this;
    }

    public function getTypeIcon(): string
    {
        return match ($this->typeOffre) {
            'emploi' => '💼',
            'projet' => '🚀',
            'tache' => '✅',
            default => '📌',
        };
    }

    public function getNbLikes(): int
    {
        return 0;
    }

    public function getNbCommentaires(): int
    {
        return 0;
    }
}

File: src/main/java/org/model/Offres/offre.java
package org.model.Offres;

import java.math.BigDecimal;
import java.time.LocalDate;

public class Offre {

    private int          idOffre;
    private String       titre;
    private String       description;
    private String       typeOffre;        // emploi | projet | tache
    private BigDecimal   budget;
    private BigDecimal   prix;
    private int          duree;            // jours
    private String       matricule;        // OFF-2024-00042

    private String       localisation;

    private LocalDate    datePublication;
    private LocalDate    dateLimite;
    private String       statut;           // ouverte | fermee
    private int          idUtilisateur;
    private String       mailUser;

    private int          vues;
    private String       tags;

    // ── Nouveaux champs likes & commentaires ───────────────────
    private int          nbLikes;
    private int          nbCommentaires;
    private boolean      likedByCurrentUser; // transient, non persisté

    // ── Constructeurs ──────────────────────────────────────────
    public Offre() {}

    public Offre(String titre, String description, String typeOffre,
                 BigDecimal budget, BigDecimal prix, int duree,
                 int nombrePlaces, String competencesReq,
                 String localisation, String modeTravail, String niveauRequis,
                 LocalDate datePublication, LocalDate dateLimite,
                 String statut, int idUtilisateur, String mailUser, String tags) {
        this.titre           = titre;
        this.description     = description;
        this.typeOffre       = typeOffre;
        this.budget          = budget;
        this.prix            = prix;
        this.duree           = duree;
        this.localisation    = localisation;
        this.datePublication = datePublication;
        this.dateLimite      = dateLimite;
        this.statut          = statut;
        this.idUtilisateur   = idUtilisateur;
        this.mailUser        = mailUser;
        this.tags            = tags;
    }

    // ── Getters / Setters ──────────────────────────────────────
    public int          getIdOffre()           { return idOffre; }
    public void         setIdOffre(int v)      { this.idOffre = v; }

    public String       getTitre()             { return titre; }
    public void         setTitre(String v)     { this.titre = v; }

    public String       getDescription()       { return description; }
    public void         setDescription(String v){ this.description = v; }

    public String       getTypeOffre()         { return typeOffre; }
    public void         setTypeOffre(String v) { this.typeOffre = v; }

    public BigDecimal   getBudget()            { return budget; }
    public void         setBudget(BigDecimal v){ this.budget = v; }

    public BigDecimal   getPrix()              { return prix; }
    public void         setPrix(BigDecimal v)  { this.prix = v; }

    public int          getDuree()             { return duree; }
    public void         setDuree(int v)        { this.duree = v; }

    public String       getMatricule()         { return matricule; }
    public void         setMatricule(String v) { this.matricule = v; }

    public String       getLocalisation()             { return localisation; }
    public void         setLocalisation(String v)     { this.localisation = v; }

    public LocalDate    getDatePublication()          { return datePublication; }
    public void         setDatePublication(LocalDate v){ this.datePublication = v; }

    public LocalDate    getDateLimite()               { return dateLimite; }
    public void         setDateLimite(LocalDate v)    { this.dateLimite = v; }

    public String       getStatut()                   { return statut; }
    public void         setStatut(String v)           { this.statut = v; }

    public int          getIdUtilisateur()            { return idUtilisateur; }
    public void         setIdUtilisateur(int v)       { this.idUtilisateur = v; }

    public String       getMailUser()                 { return mailUser; }
    public void         setMailUser(String v)         { this.mailUser = v; }

    public int          getVues()                     { return vues; }
    public void         setVues(int v)                { this.vues = v; }

    public String       getTags()                     { return tags; }
    public void         setTags(String v)             { this.tags = v; }

    // ── Nouveaux getters/setters likes & commentaires ──────────
    public int          getNbLikes()                  { return nbLikes; }
    public void         setNbLikes(int v)             { this.nbLikes = v; }

    public int          getNbCommentaires()           { return nbCommentaires; }
    public void         setNbCommentaires(int v)      { this.nbCommentaires = v; }

    public boolean      isLikedByCurrentUser()        { return likedByCurrentUser; }
    public void         setLikedByCurrentUser(boolean v){ this.likedByCurrentUser = v; }

    // ── Helpers ────────────────────────────────────────────────
    public boolean isOuverte() { return "ouverte".equalsIgnoreCase(statut); }

    public boolean isExpiree() {
        return dateLimite != null && LocalDate.now().isAfter(dateLimite);
    }

    public String getTypeIcon() {
        return switch (typeOffre == null ? "" : typeOffre) {
            case "emploi"  -> "💼";
            case "projet"  -> "🚀";
            case "tache"   -> "📋";
            default        -> "📄";
        };
    }

    @Override
    public String toString() {
        return "[" + matricule + "] " + titre + " (" + typeOffre + ") — " + prix + " TND";
    }
}